packages/meditrak-app/app/assessment/specificQuestions/FileQuestion.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * Tupaia MediTrak
+ * Copyright (c) 2017 Beyond Essential Systems Pty Ltd
+ */
+
+import React from 'react';
+import { StyleSheet, View } from 'react-native';
+import PropTypes from 'prop-types';
+import DocumentPicker from 'react-native-document-picker'
+import RNFS from 'react-native-fs';
+
+import { DEFAULT_PADDING } from '../../globalStyles';
+import { Button, Text } from '../../widgets';
+
+const renderFilePickerButton = (onPress, label) => (
+  <Button title={label} onPress={onPress} style={localStyles.button} />
+);
+
+const DumbFileQuestion = ({ onPressChooseFile, onPressRemoveFile, filename, errorMessage }) => (
+  <View style={localStyles.container}>
+    {filename !== null && <Text>{filename}</Text>}
+    {filename === null && errorMessage && <Text>{errorMessage}</Text>}
+    <View style={localStyles.actions}>
+      {renderFilePickerButton(onPressChooseFile, filename === null ? 'Attach file' : 'Change file')}
+      {filename !== null && renderFilePickerButton(onPressRemoveFile, 'Remove file')}
+    </View>
+  </View>
+);
+
+DumbFileQuestion.propTypes = {
+  onPressChooseFile: PropTypes.func.isRequired,
+  onPressRemoveFile: PropTypes.func.isRequired,
+  filename: PropTypes.string,
+  errorMessage: PropTypes.string,
+};
+
+DumbFileQuestion.defaultProps = {
+  filename: null,
+  errorMessage: null,
+};
+
+const BUTTON_SPACING = 5;
+
+const localStyles = StyleSheet.create({
+  container: {
+    marginTop: DEFAULT_PADDING,
+  },
+  actions: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    marginHorizontal: -BUTTON_SPACING,
+    flex: 1,
+  },
+  button: {
+    flex: 1,
+    marginHorizontal: BUTTON_SPACING,
+  },
+});
+
+export const FileQuestion = ({ answer, extraProps, onChangeAnswer }) => {
+  const filename = answer ? answer.substring(answer.lastIndexOf('/') + 1) : null;
+
+  const handleOpenFilePicker = async () => {
+    try {
+      // We copy the file to document storage (NSDocumentDirectory/getFilesDir on ios/android), which is our application sandboxed file storage.
```

Comment:
Interesting. So file must be copied in case the original file is not accessible at time of sync.

Reply:
Yeah reading through how files are handled on ios/android you can pick from many places, e.g. google drive. So copying means we download the file to documents for working with.